lib/modules/manager/gitlabci/extract.ts

```diff
@@ -144,8 +145,8 @@ export async function extractAllPackageFiles(
     }
 
     if (is.array(doc?.include)) {
-      for (const includeObj of doc.include) {
-        if (is.string(includeObj.local)) {
+      for (const includeObj of doc.include.filter(isGitlabIncludeLocal)) {
+        if (!is.undefined(includeObj.local)) {
```

Comment:
this condition is always true and should be removed. or move the filter to this condition instead